modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DotExpanderProcessor.java

```diff
@@ -41,10 +48,29 @@ public IngestDocument execute(IngestDocument ingestDocument) throws Exception {
             map = ingestDocument.getSourceAndMetadata();
         }
 
+        if (this.field.equals("*")) {
+            for (String key : new ArrayList<>(map.keySet())) {
```

Comment:
I don't think the keySet needs to be wrapped in a list:
```suggestion
            for (String key : map.keySet()) {
```

Reply:
This is needed to avoid a `ConcurrentModificationException`. This occurs as the `expandDot` removes the original (dotted) key from the map. That's why we need to snapshot the original keys by wrapping it into a list.